### ISSUE TITLE
Integrate file upload component into Veasy

### ISSUE THREAD

**Issue by JeremyColton** [Albert-Gao/veasy#28]
Eg can this component https://www.npmjs.com/package/react-upload-file be integrated with Veasy so it can validate file types, size, image dimensions?

**Comment by Albert-Gao** [comment]
Not really, because veasy.js depends on the `onChange`. But your suggestion is interesting and makes me think of some new potential usage. But currently, no... Sorry :)

**Comment by JeremyColton** [comment]
Ok so the upload component is separate and has it's own validation.  Can Veasy check the separate upload component's validation before setting its isFormOk variable to true?

**Comment by Albert-Gao** [comment]
There should be a trigger to trigger veasy to start validating.
The current trigger is `onChange` event, `onReset` and `onBlur`

If this upload won't trigger these events, then veasy won't validate it.

**Comment by Albert-Gao** [comment]
But, I remember there is a `DIY` to trigger the validation without those events.
You can find more information there.

Which should suits your needs. But it requires you to setup the whole thing by yourself.
Look at here for more info:
https://albert-gao.github.io/veasy/#/diy

**Comment by JeremyColton** [comment]
The File Upload component can have its own validation in its own form.
It reads the selected file and writes the base64 output to a hidden field in the Veasy form.
This Veasy field has `isRequired:true`

That should work, correct?

**Comment by Albert-Gao** [comment]
Only if it can trigger one of the 3 events.

**Comment by Albert-Gao** [comment]
`<VeasyForm>` is designed to support auto-bind for the `<form>`.

If you want to trigger the validation without those 3 events.
You need to setup by youself, which should be easy, you just need to bind all the values.

Then you can trigger the validation in a programmatic way.
Just check the above link for more info. :)

**Comment by JeremyColton** [comment]
A simpler way to trigger the onchange event on the hidden form field:
document.getElementById("hiddenField").trigger('change')

Ok?

**Comment by JeremyColton** [comment]
1. My image upload component is taken from: https://codepen.io/hartzis/pen/VvNGZP and lives in its own form.
2. Once the image has rendered, I store the base64 string in state and populate an off-screen Veasy <TextInput> with 'abc'. Populating the off-screen field with the base64 string slowed down the page, I'm guessing due to rendering issues!  This off-screen field has a `isRequired:true` validation rule in my Veasy schema. It needs any non-null value, 'abc' will suffice.
3. To then trigger Veasy validation I call this `triggerBlur(<my TextInput dom id>)`:

```
const event = new Event('blur', {
    'view': window,
    'bubbles': true,
    'cancelable': false
});

triggerBlur(domID) {
    let node = document.getElementById(domID);
    node.dispatchEvent(event);
    console.log("simluated onBlur...");
}
```

It works with React 15.6!

I have now joined my image upload to Veasy.

**Comment by Albert-Gao** [comment]
Wow, you are really a veasy lover... :D Did some amazing work!
Congrat!

May I ask what rule in veasy did you use to validate the file?

**Comment by JeremyColton** [comment]
I don't use Veasy to validate the 'file' input. I wrote my own validation that checks the selected file size. The filetype is limited on the 'file' input tag itself:

```
<input className="fileInput"
                                       type="file"
                                       onChange={(e)=>this._handleImageChange(e)}
                                       accept={this.props.allowedFileTypes}/>

The interesting part is that I can make my Veasy form indirectly dependent on the file being valid by populating an off-screen Veasy <TextInput> field based on the successful reading of the file.
```
